enos/modules/install_packages/scripts/synchronize-repos.sh

```diff
@@ -99,12 +99,26 @@ synchronize_repos() {
   esac
 }
 
+# Function to check cloud-init status and retry on failure
 # Before we start to modify repositories and install packages we'll wait for cloud-init to finish
 # so it doesn't race with any of our package installations.
-# We run as sudo becase Amazon Linux 2 throws Python 2.7 errors when running `cloud-init status` as
+# We run as sudo because Amazon Linux 2 throws Python 2.7 errors when running `cloud-init status` as
 # non-root user (known bug).
-sudo cloud-init status --wait
+check_cloud_init() {
```

Comment:
I would say that our motivation here is to not to check cloud init, rather to wait until it's done so that cloud-init and us are not racing to update/modify either repository config, databases, or packages. Perhaps `wait_for_cloud_init()`?